proxy/src/config.rs

```diff
@@ -299,23 +302,35 @@ impl<'a> TryFrom<&'a Strings> for Config {
         let tls_settings = match (tls_trust_anchors?,
                                   tls_end_entity_cert?,
                                   tls_private_key?,
-                                  pod_name?.as_ref())
+                                  tls_pod_identity_template?.as_ref(),
+                                  tls_controller_identity?)
         {
             (Some(trust_anchors),
              Some(end_entity_cert),
              Some(private_key),
-             Some(pod_name)) => {
-                let service_identity = tls::Identity::try_from_pod_name(&namespaces, pod_name)
+             Some(tls_pod_identity_template),
+             controller_identity) => {
+                let pod_identity =
+                    tls_pod_identity_template.replace(VAR_POD_NAMESPACE, &namespaces.pod);
```

Comment:
Before doing this, should we validate that `tls_pod_identity_template` actually contains `VAR_POD_NAMESPACE`?

Reply:
> Before doing this, should we validate that tls_pod_identity_template actually contains VAR_POD_NAMESPACE?

If/when the proxy is used outside of Kubernetes, there might not be the need for any substitution, so I think that `String::replace()` being a no-op when there's no match is the right semantics already.

Reply:
Ah, yes, that makes sense. Perhaps eventually we might want to do that validation only if the pod namespace environment variable is set?

Reply:
> Ah, yes, that makes sense. Perhaps eventually we might want to do that validation only if the pod namespace environment variable is set?

I'm not convinced we need to validate it. I would rather we have the flexibility to switch to a different scheme that doesn't necessarily require this kind of substitution. For example, if/when injection is done using an admission controller, the namespace will be available in the injector, and then we wouldn't need this variable substitution.